### PR TITLE
feat(storage): add conf test scenario testing downloads handle complex retries

### PR DESCRIPTION
**Pull request by cojenco**
This proposes adding scenario 8 that tests retrying an interrupted download. If a retryable error occurs mid-download, the download retries and starts sending data from where it left off ( `offset_of_last_byte_received`)

Note 1: Data integrity assertions should be added to "storage.objects.get" methods conf tests so data corruption issues can be captured if any
Note 2: If agreed upon the team, I'll update the retry conformance test design doc
Note 3: Scenario 7 is WIP, and a placeholder is added

- cases:  ["return-broken-stream", "return-broken-stream"]
   (returns a broken stream after a few bytes)
- methods: "storage.objects.get"

### CODE REVIEW

File: storage/v1/retry_tests.json [hunk 1]
```diff
@@ -239,6 +239,28 @@
       ],
       "preconditionProvided": true,
       "expectSuccess": false
+    },
+    {
+      "id": 7,
+      "description": "resumabe_uploads_handle_complex_retries",
```
Comment (tritone) [inline storage/v1/retry_tests.json:245]: typo in "resumable"

File: storage/v1/retry_tests.json [hunk 2]
```diff
@@ -239,6 +239,28 @@
...
+      "methods": [],
+      "preconditionProvided": true,
+      "expectSuccess": true
+    },
+    {
+      "id": 8,
+      "description": "downloads_handle_complex_retries",
+      "cases": [
+        {
+          "instructions": ["return-broken-stream", "return-broken-stream"]
```
Comment (tritone) [inline storage/v1/retry_tests.json:256]: Do we plan on adding `return-XXX-after-YYYk` to this as well?

File: storage/v1/retry_tests.json [hunk 3]
```diff
@@ -239,6 +239,28 @@
...
+        }
+      ],
+      "methods": [
+        {"name": "storage.objects.get",                   "resources": ["BUCKET", "OBJECT"]}
```
Comment (tritone) [inline storage/v1/retry_tests.json:260]: How can we filter this just to downloads vs. metadata GET operations? Or are we okay with just having extra tests run on these operations unnecessarily.

File: storage/v1/retry_tests.json (hunk 2 above)
Comment (cojenco) [reply storage/v1/retry_tests.json:256]: Yep that's the plan once `return-XXX-after-YYYk` is supported for downloads. If we add it now, the tests will break.

File: storage/v1/retry_tests.json (hunk 3 above)
Comment (cojenco) [reply storage/v1/retry_tests.json:260]: Great question! The main factor will probably be if and how the storage testbench can differentiate a download vs. metadata get. Right now the testbench only recognizes the method as `storage.objects.get`. If there's no straightforward way to filter downloads vs metadata GET in the testbench, I think it's okay with having extra tests run on metadata GET operations. Let's give this some more thought ;)

File: storage/v1/retry_tests.json [hunk 4]
```diff
@@ -239,6 +239,28 @@
...
```
Comment (BenWhitehead) [inline storage/v1/retry_tests.json:260]: Wasn't there going to be a group added to the schema here?

**Comment by BenWhitehead** [comment]
Also what's the relation between this PR and https://github.com/googleapis/conformance-tests/pull/67?

File: storage/v1/retry_tests.json (hunk 4 above)
Comment (cojenco) [reply storage/v1/retry_tests.json:260]: @BenWhitehead Closing this PR by replacement #67. Sorry for the confusion. The go build is failing. I triggered a new build this morning, awaiting for it to finish.

**Comment by cojenco** [comment]
Closing the PR with replacement #67

**Pull request closed without merging**